### ISSUE TITLE
getting the same share for both parties with the python mpc script

### ISSUE THREAD

**Issue by Tony-Stark** [unbound-tech/blockchain-crypto-mpc#22]
I followed the comments in mpc_demo.py to test key generation. The example is as follows.
Example 1: Generate a split EDDSA key
    user1@host1> python mpc_demo.py --out_file key_share.bin --server
    user2@host2> python mpc_demo.py --type EDDSA --command generate --out_file key_share.bin --host host1

I ran them both on the same machine, but in two different sessions kinda like this:
    user1@host1> python mpc_demo.py --out_file key_share1.bin --server
    user1@host1> python mpc_demo.py --type EDDSA --command generate --out_file key_share2.bin --host localhost
    
but key_share1 and key_share2 are identitcal. 
What's happening? Is it because it runs on the same machine? Or did I misunderstood something else?

**Comment by Tony-Stark** [comment]
I tried to check the python script (the same way as above) using two machines instead of localhost. I checked the firewall rules, and checked that the correct ports are up on both sides. The 2 machines communicate fine with other programs, and there's a ping. But using the python script nothing happens for awhile and then I get
Traceback (most recent call last):
  File "mpc_demo.py", line 273, in <module>
    run_client()
  File "mpc_demo.py", line 216, in run_client
    clientsocket.connect((args.host, args.port))
TimeoutError: [Errno 110] Connection timed out
